File: Greedy Algorithm/435.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        if(intervals.empty()){
            return 0;
        }
        int len = intervals.size();
        sort(intervals.begin(), intervals.end(), [](vector<int>&a, vector<int>&b) {
            return a[1] < b[1]; // sort by the second element in each vector
            // do not forget & & 
        });
        int prev = intervals[0][1];
        int sum = 0;
        for(int i{1}; i< len; ++i){

            if (prev > intervals[i][0]){
                sum++;
            }else{
                prev = intervals[i][1];
            }
        }
        return sum;
    }
};

File: Greedy Algorithm/435.py
class Solution(object):
    def eraseOverlapIntervals(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: int
        """
        intervals.sort()
        sum = 0
        prev = intervals[0][1]
        for i in range (1,len(intervals)):
            prev = min(intervals[i][1], prev)
            if prev > intervals[i][0]:
                sum += 1
            else:
                prev = intervals[i][1]
        return(sum)
